src/client/mod.rs

```diff
@@ -157,41 +157,51 @@ impl APIClient {
         trace!("{} {}", res.status().as_str(), res.url());
 
         // Now use `unfold` to convert the chunked responses into a Stream
-        Ok(futures::stream::unfold(res, |mut resp| {
-            async move {
-                // A buffer to store our current object in
-                // as we stream it's entirety
-                let mut buff = Vec::new();
-
-                loop {
-                    match resp.chunk().await {
-                        Ok(Some(chunk)) => {
-                            // append it to our current buffer
-                            buff.extend_from_slice(&chunk);
-
-                            // Check to see if we have valid JSON in our buffer,
-                            // and if so parse it fully and return.
-                            // Deserializing as serde::de::IgnoredAny is just a syntax check
-                            // and does not go through any of the memory intensity of constructing the actual
-                            // value.
-                            let valid = serde_json::from_slice::<IgnoredAny>(&buff);
-
-                            if valid.is_ok() {
-                                match serde_json::from_slice::<T>(&buff) {
-                                    Ok(val) => return Some((Ok(val), resp)),
-                                    Err(e) => {
-                                        warn!("{} {:?}", String::from_utf8_lossy(&buff), e);
-                                        return Some((Err(Error::SerdeError(e)), resp));
+        Ok(futures::stream::unfold(
+            (res, Vec::new(), Vec::new()),
+            |(mut resp, mut buff, mut items): (_, _, Vec<Result<T>>)| {
+                async {
+                    // If we have any items, pop off the first,
+                    // yield it, and then continue into the next iteration
+                    if !items.is_empty() {
+                        let current = items.pop().unwrap(); // We know items is not empty so this is safe
+                        return Some((current, (resp, buff, items)));
+                    }
+
+                    loop {
+                        match resp.chunk().await {
+                            Ok(Some(chunk)) => {
+                                // append it to our current buffer
+                                buff.extend_from_slice(&chunk);
+
+                                if chunk.contains(&b'\n') {
+                                    let mut new_buff = Vec::new();
+                                    let mut new_items = Vec::new();
+
+                                    for line in buff.split(|x| x == &b'\n') {
+                                        match serde_json::from_slice(&line) {
+                                            Ok(val) => new_items.push(Ok(val)),
```

Comment:
Oh, is it because you can only return a single element from this call? You might be able to use [concat](https://docs.rs/futures/0.3.1/futures/stream/trait.StreamExt.html#method.concat) from futures to avoid this problem.

Reply:
Yeah this was exactly why, I'll look into `concat` that looks promising

Reply:
Hmm looks like `concat` will consume a `Stream<Item = T>` had convert it to like a `Future<Item = Vec<T>>`, so that won't do it. What I'm working on instead is first constructing a `Stream<Item = Vec<Result<T>>` that is allowed to yield multiple values per iteration, and then using `StreamExt::flatten` to get us to the `Stream<Item = Result<T>>` we want.